Libraries/src/Amazon.Lambda.Annotations.SourceGenerators/LambdaFunctionCodeGenerator.cs

```diff
@@ -0,0 +1,261 @@
+                        var methodSymbol = classModel.GetDeclaredSymbol(methodSyntax);
+                        if (methodSymbol != null
+                            && methodSymbol.Parameters.Count() == 1
+                            && methodSymbol.Parameters[0].Type.Equals(iServiceCollectionSymbol, SymbolEqualityComparer.Default))
+                        {
+                            if (_configureMethodSymbol != null)
+                            {
+                                throw new NotSupportedException(
+                                    "Multiple LambdaStartup classes are not allowed in a Lambda project.");
+                            }
+
+                            _configureMethodSymbol = methodSymbol;
+                            break;
+                        }
+                    }
+                }
+
+                return _configureMethodSymbol;
+            }
+        }
+
+        /// <summary>
+        /// Returns whether the simplified Lambda function returns void or <see cref="System.Threading.Tasks.Task"/>.
+        /// </summary>
+        public bool LambdaFunctionReturnsVoidOrTask
+        {
+            get
+            {
+                if (LambdaFunctionMethodSymbol.ReturnsVoid)
+                {
+                    return true;
+                }
+
+                var taskSymbol = _context.Compilation.GetTypeByMetadataName("System.Threading.Tasks.Task");
+                if (LambdaFunctionMethodSymbol.ReturnType.Equals(taskSymbol, SymbolEqualityComparer.Default))
+                {
+                    return true;
+                }
+
+                return false;
+            }
+        }
+
+        public LambdaFunctionCodeGenerator(
+            MethodDeclarationSyntax lambdaFunctionMethodSyntax,
+            ClassDeclarationSyntax startupSyntax,
+            GeneratorExecutionContext context)
+        {
+            _lambdaFunctionMethodSyntax = lambdaFunctionMethodSyntax;
+            _startupSyntax = startupSyntax;
+            _context = context;
+        }
+
+        /// <summary>
+        /// Generates <see cref="SourceText"/> for the LambdaFunction including setting up the dependency injection.
+        /// </summary>
+        /// <returns>A <see cref="Tuple"/> containing source file name hint and source text.</returns>
+        public (string, SourceText) GenerateSource()
+        {
+            var apiGatewayResponseSymbol =
+                _context.Compilation.GetTypeByMetadataName("Amazon.Lambda.APIGatewayEvents.APIGatewayProxyResponse");
+
+            var returnTypeString = "APIGatewayProxyResponse";
+            if (LambdaFunctionMethodSymbol.IsAsync)
+            {
+                returnTypeString = $"System.Threading.Tasks.Task<{returnTypeString}>";
+            }
+
+            var source = new StringBuilder(
+$@"using System;
+using Amazon.Lambda.Core;
+using Amazon.Lambda.APIGatewayEvents;
+using Microsoft.Extensions.DependencyInjection;
```

Comment:
Is it going to be later task to handle a Lambda function that isn't using DI and in that case doesn't have a package reference to Microsoft.Extensions.DependencyInjection?

Reply:
I think DI is must, Startup is optional. This will always be there. We inject customer code as scoped and resolve later, which needs Microsoft.Extensions.DependencyInjection package.